src/services/zelcoreRates.js

```diff
@@ -14,89 +14,47 @@ function apiRequest(url) {
+      apiRequest('https://min-api.cryptocompare.com/data/pricemulti?fsyms=HUSH,ZCL,XSG,BTCP,ZEN,KMD,XZC,ZER,ABT,ADX,AE,AION,AST,BBO,APPC,BLZ,BNT,ETHOS,COFI,DAI,DGX,ELEC,ELF,ENJ,STORJ,IOST,DENT,LEND,LINK,MANA,LRC,QASH,ICN,MCO,MTL,POE,POLY,POWR,RCN,RDN,REQ,SNT,SALT,STORM,EDO,TUSD,DCN,WAX,WINGS,DTA,FUN,KIN,BSV,AOA,THETA,ADT,MFT,ATL,ANT,ARN,BRD,REP,QKC,LOOM,ANON,EURS,AMB,BCPT&tsyms=BTC'), //  2
+      apiRequest('https://min-api.cryptocompare.com/data/pricemulti?fsyms=EOS,ADA,XRP,DOCK,NEO,TRON,BTT,SAFE,SUQA,BTH,GRS&tsyms=BTC'),  //  3
+      apiRequest('https://api.coingecko.com/api/v3/coins/markets?vs_currency=btc&ids=safe-coin-2,genesis-network,bzedge,commercium,bitcoin-zero,zelcash&order=market_cap_desc&per_page=100&page=1&sparkline=false'),  // 4
     ]).then((results) => {
       var rates = [];
       var efg = {};
       var errors = { errors: {} }
-      var ccDataA = results[0]; // results from cryptocompare
-      var ccDataB = results[1]; // results from cryptocompare
-      var ccDataC = results[2]; // results from cryptocompare
-
+      
+      // results from bitpay (fiat rates)
       try {
-        var dummyTest = results[3][1].code; // results from bitpay
-        var bitpayData = results[3]
+        var dummyTest = results[0][1].code;
+        var bitpayData = results[0]
       } catch (e) {
         var bitpayData = -1
-        errors.errors.bitPayData = results[3]
+        errors.errors.bitPayData = results[0]
       }
 
-      try {
-        var safeprice = Number(results[4][0].current_price) // results from coingecko
-        efg.SAFE = safeprice
-      } catch (e) {
-        errors.errors.SAFE = results[4]
-      }
-      try {
-        var genxprice = Number(results[5][0].current_price) // results from coingecko
-        efg.GENX = genxprice
-      } catch (e) {
-        errors.errors.GENX = results[5]
-      }
-      try {
-        var bzeprice = Number(results[6][0].current_price) // results from coingecko
-        efg.BZE = bzeprice
-      } catch (e) {
-        errors.errors.BZE = results[6]
-      }
-      try {
-        var cmmprice = Number(results[7][0].current_price) // results from coingecko
-        efg.CMM = cmmprice
-      } catch (e) {
-        errors.errors.CMM = results[7]
-      }
-      try {
-        var bzxprice = Number(results[8][0].current_price) // results from coingecko
-        efg.BZX = bzxprice
-      } catch (e) {
-        errors.errors.BZX = results[8]
-      }
-
-      var coinsA = Object.keys(ccDataA) // results from cryptocompare
-      coinsA.forEach((coin) => {
+      // results from cryptocompare (prices)
+      var coinsCC = Object.keys(
+	{...results[1],
```

Comment:
The spreading is not considering an errored result. 
Let's assume some of the request failes, then we are putting inccorrect values here.
I think current implementation is not considering failed results neither and thus incorrect coins array may contain wrong stuff. Should be corrected.

Reply:
Fixed in new commit.